beacon-chain/blockchain/service.go

```diff
@@ -300,11 +302,17 @@ func (c *ChainService) blockProcessing(done <-chan struct{}) {
 				log.Debugf("Incoming block failed validity conditions: %v", err)
 			}
 
-			// Process attestation as a validator.
-			if err := c.chain.processAttestations(block); err != nil {
-				// We might receive a lot of blocks that fail attestation processing,
-				// so we create a debug level log instead of an error log.
-				log.Debugf("could not process attestation: %v", err)
+			// Process attestations as a beacon chain node.
+			for index := range block.Attestations() {
+				if err := c.chain.processAttestation(index, block); err != nil {
+					// We might receive a lot of blocks that fail attestation processing,
+					// so we create a debug level log instead of an error log.
+					log.Debugf("could not process attestation: %v", err)
+				}
+				blockVoteCache, err = c.chain.calculateBlockVoteCache(index, block)
```

Comment:
`calculateBlockVoteCache` would be more clear if it accepted the cache and returned the updated one. As of now, it's returning `blockVoteCache` but this loop is doing nothing with it until the last iteration.

Reply:
Ok, I'll move this out side of attestation loop

Reply:
If processAttestation passes then we update voteCache